econml/_shap.py

```diff
@@ -144,12 +150,12 @@ def _shap_explain_model_cate(cme_model, models, X, d_t, d_y, feature_names=None,
         except Exception as e:
             print("Final model can't be parsed, explain const_marginal_effect() instead!", repr(e))
             return _shap_explain_cme(cme_model, X, d_t_, d_y_,
-                                     feature_names=feature_names_,
+                                     feature_names=None,
```

Comment:
why aren't we passing the un-featureized feature names here??

Reply:
Yes that's the plan. My concern is we have to keep track too many names here. First of all, the logic is we have to check whether the user inputs a feature name, if not see if we could get feature name from `._input_names`, then we have to save those names for both un-featurized names and featurized names and pick the right one when shap explain final model or `const_marginal_effect`. I will follow this logic unless we find a way to simplify it.

Reply:
I think for each shap helper function (except for `_shap_explain_cme`), I will input raw X, fitted featurizer, input name dict, cate_feature_names function, and keep both the raw feature names and featurized feature names from`_define_names`  helper function. It's also needed to be done once Miruna's PR is merged.

Currently there is a bug if users don't specify feature_names, we use the feature_names from _input_names, but it's un-featurized...